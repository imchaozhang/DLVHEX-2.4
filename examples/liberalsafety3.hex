% The program is safe because testSmallerThan binds Len, which binds in turn Y and X

s(a).
s(Y) :- s(X), &testConcat[X, a](Y), &testStrlen[Y](Len), &testSmallerThan[Len, 10]()<finitefiber>.
