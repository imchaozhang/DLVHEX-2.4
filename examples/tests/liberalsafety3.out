{s(a),s(aa),s(aaa),s(aaaa),s(aaaaa),s(aaaaaa),s(aaaaaaa),s(aaaaaaaa),s(aaaaaaaaa)}
